src/pages/Auth/Login.jsx

```diff
@@ -0,0 +1,103 @@
+import { MdEmail } from "react-icons/md";
+import { RiLockPasswordFill } from "react-icons/ri";
+import { AiFillEye, AiFillEyeInvisible } from "react-icons/ai";
+import "./Auth.css"
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { useAuth } from "../../contexts";
+import { useEffect } from "react";
+import { ACTION_TYPE } from "../../utils";
+import { Header } from "../../components";
+
+export const Login = () => {
+
+    // Auth context values here 
+    const { loginFormHandler, authState: { error, authToken }, authDispatch } = useAuth();
+    const { SET_ERROR } = ACTION_TYPE
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    // Formdata state here
+    const [loginFormData, setloginFormData] = useState({
+        email: "",
+        password: "",
+        passType: "password"
+    })
+    const [rememberMe, setRememberMe] = useState(false);
+
+    // On submit handler 
+    const formSubmitHandler = (event) => {
+        event.preventDefault();
+        loginFormHandler(loginFormData, rememberMe);
+    }
+
+    // To fill test data 
+    const testCredentialHandler = () => {
+        setloginFormData({
+            email: "sudo123@gmail.com",
+            password: "sudo123",
+            passType: "password",
+        })
+    }
+
+    // password visibilty handler here 
+    const passVisibiltyHandler = () => {
+        return loginFormData.passType === "password" ? setloginFormData({ ...loginFormData, passType: "text" }) : setloginFormData({ ...loginFormData, passType: "password" });
+    }
+
+    // Checking if alredy login
+    useEffect(() => {
+        // 
+        if (authToken) { navigate(`${location?.state?.from?.pathname}`) }
```

Comment:
give threshold value like '/' , what if you location.state.from is undefined

Reply:
Missed that. Fixed now.